great_expectations/expectations/metrics/util.py

```diff
@@ -258,12 +250,15 @@ def _get_dialect_type_module(dialect=None):
             "No sqlalchemy dialect found; relying in top-level sqlalchemy types."
         )
         return sa
-    try:
-        # Redshift does not (yet) export types to top level; only recognize base SA types
-        # noinspection PyUnresolvedReferences
-        if isinstance(dialect, sqlalchemy_redshift.dialect.RedshiftDialect):
-            return dialect.sa
-    except (TypeError, AttributeError):
+
+    # Redshift does not (yet) export types to top level; only recognize base SA types
+    # noinspection PyUnresolvedReferences
+    if aws.redshiftdialect and isinstance(
+        dialect,
+        aws.redshiftdialect.RedshiftDialect,
+    ):
+        return dialect.sa
+    else:
         pass
```

Comment:
nit (non-blocking): I don't think the `else pass` is required

```suggestion
```

Reply:
@TrangPham Great catch -- deleted.  Thanks!